File: Functions/update_budget.py

print("Hello and welcome to your finacial calculator")
income = float(input("What is your montly income?:\n"))
rent = float(input("What is your monthly rent?:\n"))
utilitties = float(input("What is your monthly utilitties?:\n"))
groceries = float(input("What is your monthly groceries?:\n"))
transp = float(input("What is your monthly transportation cost?:\n"))
savings = income * .2
expenses = rent+utilitties+groceries+transp
spending = income-expenses-savings
def percent(type, amount):
    per = amount/income *100

    return f"your {type} is {per}% income."

print(f"Your monthly income is ${income:.2f}\n")
print(f"Your monthly expenses are ${expenses:.2f}\n")
print(f"Your monthly savings is ${savings:.2f}\n")
print(f"Your monthly spending money is ${spending:.2f}\n")
print(percent("rent", rent))
print(percent("utilities", utilitties))
print(percent("groceries", groceries))
print(percent("transportation", transp))
print(percent("savings", savings))
print(percent("expenses", expenses))

File: Functions/update_budget.c
#include <stdio.h>
float income, rent, utilities, groceries, transportation, savings, expenses, spend;

float input(char type[], float var){
    printf("Monthly %s:\n", type);
    scanf("%f", &var);
    return var;
}


void percent(char type[], int amount){
    int per = amount/income *100;

    printf("Your %s type is %d%% of your income.\n", type, per);
}


int main(void){
    printf("This is a budget calculator.\n");
    income = input("income", income);
    rent = input("rent", rent);
    utilities = input("utilities", utilities);
    groceries = input("groceries", groceries);
    transportation = input("transportation", transportation);

    savings = income * .2;
    expenses = rent + utilities + groceries + transportation;
    spend = income - expenses - savings;
    printf("You make $%.2f\n", income);
    printf("You expenses are $%.2f\n", expenses);
    printf("You savings are $%.2f\n", savings);
    printf("You spending money is $%.2f\n", spend);

    percent("rent", rent);
    percent("utilities", utilities);
    percent("groceries", groceries);
    percent("transportation", transportation);
    percent("expenses", expenses);
    percent("savings", savings);
    percent("spend", spend);

    return 0;
}